src/core/ai/llm/registry.ts

```diff
@@ -280,6 +347,202 @@ export function isValidProvider(provider: string): boolean {
     return provider.toLowerCase() in LLM_REGISTRY;
 }
 
+/**
+ * Gets the supported file types for a given provider.
+ * @param provider The name of the provider.
+ * @returns Array of supported file types for the provider, or empty array if provider not found.
+ */
+export function getSupportedFileTypes(provider: string): SupportedFileType[] {
```

Comment:
i don't think we need provider level functions or defaults

let's keep it at model level only, generalizing across a provider can be unreliable, some models for that provider may support, some models may not.

can remove all the functions that accept only provider and not model